GVRf/Framework/jni/objects/components/camera_rig_jni.cpp

```diff
@@ -126,49 +120,24 @@ Java_org_gearvrf_NativeCameraRig_getLookAt(JNIEnv * env,
 JNIEXPORT jlong JNICALL
 Java_org_gearvrf_NativeCameraRig_ctor(JNIEnv * env,
         jobject obj) {
-    std::shared_ptr<CameraRig> camera_rig(new CameraRig());
-    return reinterpret_cast<jlong>(new std::shared_ptr<CameraRig>(camera_rig));
+    return reinterpret_cast<jlong>(new CameraRig());
 }
 
 JNIEXPORT jint JNICALL
 Java_org_gearvrf_NativeCameraRig_getCameraRigType(JNIEnv * env,
         jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
+    CameraRig* camera_rig = reinterpret_cast<CameraRig*>(jcamera_rig);
     return static_cast<jint>(camera_rig->camera_rig_type());
 }
 
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeCameraRig_setCameraRigType(JNIEnv * env,
         jobject obj, jlong jcamera_rig, jint camera_rig_type) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
+    CameraRig* camera_rig = reinterpret_cast<CameraRig*>(jcamera_rig);
     camera_rig->set_camera_rig_type(
             static_cast<CameraRig::CameraRigType>(camera_rig_type));
 }
 
-JNIEXPORT jlong JNICALL
-Java_org_gearvrf_NativeCameraRig_getLeftCamera(JNIEnv * env,
-        jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
-    std::shared_ptr<Camera> left_camera(camera_rig->left_camera());
-    return left_camera.get() == NULL ?
-            0 :
-            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(left_camera));
-}
-
-JNIEXPORT jlong JNICALL
-Java_org_gearvrf_NativeCameraRig_getRightCamera(JNIEnv * env,
-        jobject obj, jlong jcamera_rig) {
-    std::shared_ptr<CameraRig> camera_rig = *reinterpret_cast<std::shared_ptr<
-            CameraRig>*>(jcamera_rig);
-    std::shared_ptr<Camera> right_camera(camera_rig->right_camera());
-    return right_camera.get() == NULL ?
-            0 :
-            reinterpret_cast<jlong>(new std::shared_ptr<Camera>(right_camera));
-}
-
```

Comment:
>  right now I am thinking that actually there is nothing internal that is using camera_rig::left/right_camera() and also, the developer may never reach these two methods either

Looks like you're right: No references to `CameraRig::left_camera()` or `CameraRig::right_camera()`. Otoh, cleaning out dead code was _not_ the point of this checkin - I only zapped jni methods that weren't being called any longer.